www/js/control/uploadService.ts

```diff
@@ -0,0 +1,131 @@
+import { logDebug, logInfo, logError, displayError, displayErrorMsg } from "../plugin/logger";
+import i18next from "i18next";
+
+/**
+ * @returns A promise that resolves with an upload URL or rejects with an error
+ */
+async function getUploadConfig() {
+    return new Promise<string[]>(async function (resolve, reject) {
+        logInfo( "About to get email config");
+        let url = [];
+        try {
+            let response = await fetch("json/uploadConfig.json");
+            let uploadConfig = await response.json();
+            logDebug("uploadConfigString = " + JSON.stringify(uploadConfig['url']));
+            url.push(uploadConfig["url"]);
+            resolve(url);
+        } catch (err) {
+            try{
+                let response = await fetch("json/uploadConfig.json.sample");
+                let uploadConfig = await response.json();
+                logDebug("default uploadConfigString = " + JSON.stringify(uploadConfig['url']));
+                console.log("default uploadConfigString = " + JSON.stringify(uploadConfig['url']))
+                url.push(uploadConfig["url"]);
+                resolve(url);
+            } catch (err) {
+                logError("Error while reading default upload config" + err);
+                reject(err);
+            }
+        } 
+    })
+}
```

Comment:
To keep you in the loop- there is an ongoing discussion that may be relevant to this.
https://github.com/e-mission/e-mission-docs/issues/985#issuecomment-1760270148

I see that this looks for a file called `uploadConfig.json`. For the email service, there is similarly the `emailConfig.json` file. I am proposing that we move away from these individual JSON files and instead support them as optional fields in the dynamic config. And if not present in the dynamic config, we'll fallback the default which would be hardcoded here

Reply:
Good to know, I think it does make sense to configure this with all the other options. Right now it's just a single URL that is needed. If we decide to make that change now I can definitely include it here, else I can come back and clean it up when we do make the change.

Reply:
Let's handle the app-level config separately, and include this in there
https://github.com/e-mission/e-mission-docs/issues/985#issuecomment-1769790309

This is fine for now.